packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -43,9 +44,36 @@ const decorator = (service) => ({
     const data = { ...opts.data };
     if (isNull(data[ENTITY_STAGE_ATTRIBUTE])) {
       delete data[ENTITY_STAGE_ATTRIBUTE];
+      return service.update.call(this, uid, entityId, { ...opts, data });
     }
 
-    return service.update.call(this, uid, entityId, { ...opts, data });
+    const entity = await service.findOne.call(this, uid, entityId, {
+      populate: {
+        [ENTITY_STAGE_ATTRIBUTE]: {
+          populate: {
+            workflow: true,
+          },
+        },
+      },
+    });
+    const previousStageId = entity?.[ENTITY_STAGE_ATTRIBUTE]?.id ?? null;
+
+    const updatedEntity = await service.update.call(this, uid, entityId, { ...opts, data });
+    if (previousStageId && previousStageId !== data[ENTITY_STAGE_ATTRIBUTE]) {
+      const webhookPayload = {
+        entityId,
+        workflow: {
+          id: entity[ENTITY_STAGE_ATTRIBUTE].workflow.id,
+          stages: {
+            from: previousStageId,
+            to: data[ENTITY_STAGE_ATTRIBUTE],
+          },
+        },
+      };
+      await service.emitEvent.call(this, uid, WORKFLOW_UPDATE_STAGE, webhookPayload);
+    }
```

Comment:
Should we only emit the event if it's configured to do so in the Webhooks settings page?

Reply:
Do you mean only if there is a webhook defined for the `WORKFLOW_UPDATE_STAGE` event?

I think the approach we use elsewhere is to emit the events regardless of whether a webhook exists for this event and then the event hub deals with actually sending the webhooks

Reply:
> Do you mean only if there is a webhook defined for the WORKFLOW_UPDATE_STAGE event?

I mean if the "Stage Change" setting is enabled or not
![image](https://github.com/strapi/strapi/assets/20578351/a0f1b7bd-f141-4f8a-8be4-fbd9d6e3bfbd)


Because in this PR the event is sent regardless of this setting configuration.

Reply:
Yes I think that's fine. The events should still be emitted but they will only be picked up by webhooks defined for this event type.

e.g. if none are defined nothing will happen

Reply:
I think we don't want to send any webhook to that configured webhook if "Review Workflows Stage Change" is not enabled no? I am receiving the event on the webhook side even when its disabled.

Reply:
oh that might be it, you are right 💡